rack-protection/lib/rack/protection/base.rb

```diff
@@ -93,7 +93,10 @@ def session(env)
       end
 
       def drop_session(env)
-        session(env).clear if session? env
+        return unless session? env
+
+        warn env, "session dropped by #{self.class}" unless ENV["RACK_PROTECTION_SILENCE_REACTIONS"] == "1"
+        session(env).clear
```

Comment:
This is what I had in mind :) (wasn't by a computer when doing earlier comments)

```suggestion

        session(env).clear
        
        return if ["1", "true"].include?(ENV["RACK_PROTECTION_SILENCE_DROP_SESSION_WARNING"])
        
        warn env, "session dropped by #{self.class}"
```

Reply:
OK, I've squashed that in

Reply:
OK, I've squashed that in